### ISSUE TITLE
Infinite loading when using menyoo in FiveM story mode

### ISSUE THREAD

**Issue by BoGnY** [itsjustcurtis/MenyooSP#86]
Infinite loading when using menyoo in FiveM story mode.

GTA version: 3179
FiveM version: 2699

With original menyoo 1.8.1 all works correctly.

**Comment by itsjustcurtis** [comment]
You're on a very outdated version of fivem that new menyoo is not compatible with.

**Comment by BoGnY** [comment]
how can be possible that just downloaded fivem setup can be outdated?

**Comment by itsjustcurtis** [comment]
Ask cfx, your version is 2699, the current requirements are 3095 or newer.

**Comment by BoGnY** [comment]
problem persist in 3095 too

**Comment by TayMcKenzieNZ** [comment]
The ASI file should in theory support all gamebuilds. That's at least, what I used to do with MAFINS version. It's fine in a server but infinite loading in FiveM storymode.

**Comment by BoGnY** [comment]
yes and problem is on some commit in this repo because original 1.8.1 works on fivem story mode

EDIT: I've tried all versions, latest working is 2.0.0b2... starting from 2.0.0rc story mode don't work

**Comment by itsjustcurtis** [comment]
Interesting that it was the change to 2.0.0rc that appears to break. That is where a number of memory changes were made, I am thinking that Fivem is perhaps not equipped for this new method of memory interrogation. 

With that in mind, going to reopen this as a new issue as it could do with some investigation.

**Comment by BoGnY** [comment]
if you need tester, I'm available 👍🏻

**Comment by itsjustcurtis** [comment]
Not heard a repeat report of this issue with lots of FiveM players still using Menyoo. Closing this issue.

**Comment by lucienlmy** [comment]
> Infinite loading when using menyoo in FiveM story mode.
> 
> GTA version: 3179 FiveM version: 2699
> 
> With original menyoo 1.8.1 all works correctly.

bro，Can the story mode of FiveM load asi scripts? I was wondering why my FiveM always gets stuck in an infinite loading screen when entering the game. Could it be a problem with Menyoo?

**Comment by itsjustcurtis** [comment]
> > Infinite loading when using menyoo in FiveM story mode.
> > 
> > GTA version: 3179 FiveM version: 2699
> > 
> > With original menyoo 1.8.1 all works correctly.
> 
> bro，Can the story mode of FiveM load asi scripts? I was wondering why my FiveM always gets stuck in an infinite loading screen when entering the game. Could it be a problem with Menyoo?

Asi scripts work fine in FiveM. The latest release includes install instructions for FiveM. Scripts need to be enabled in the server config file for them to work.

**Comment by lucienlmy** [comment]
> > > Infinite loading when using menyoo in FiveM story mode.
> > > GTA version: 3179 FiveM version: 2699
> > > With original menyoo 1.8.1 all works correctly.
> > 
> > 
> > bro，Can the story mode of FiveM load asi scripts? I was wondering why my FiveM always gets stuck in an infinite loading screen when entering the game. Could it be a problem with Menyoo?
> 
> Asi scripts work fine in FiveM. The latest release includes install instructions for FiveM. Scripts need to be enabled in the server config file for them to work.

I've been troubled by the infinite loading of FiveM's story mode for several months. I always thought it was a problem with the FiveM update. Until I saw this issue, I just tried removing Menyoo, and I could indeed enter the game. FiveM's story mode is not a server and has nothing to do with the server version. Menyoo can run on the server, but it's only in FiveM's story mode that it causes infinite loading. It's definitely a problem with the Menyoo script.

**Comment by itsjustcurtis** [comment]
Unfortunately all I can recommend is using the retail single player game. Menyoo was not designed for FiveM, it just happens to have compatibility with it. If it is not working with the story mode version of FiveM, then there is little that can be done about that without support from a FiveM specialist.

**Comment by lucienlmy** [comment]
> Unfortunately all I can recommend is using the retail single player game. Menyoo was not designed for FiveM, it just happens to have compatibility with it. If it is not working with the story mode version of FiveM, then there is little that can be done about that without support from a FiveM specialist.

The original submitter mentioned that it worked normally in version 1.8.1. I think if you feel like it, you can test it with different versions to see which commit caused the issue.

**Comment by itsjustcurtis** [comment]
They also mentioned that it stopped working in 2.0.0rc, if you can confirm
earlier alpha and beta builds work for you, then I may be able to see what
the difference is.


Curtis Elliott

**Comment by lucienlmy** [comment]
> They also mentioned that it stopped working in 2.0.0rc, if you can confirm
> earlier alpha and beta builds work for you, then I may be able to see what
> the difference is.
> 
> 
> Curtis Elliott
> […](#)

I'll test it later.

**Comment by lucienlmy** [comment]
> They also mentioned that it stopped working in 2.0.0rc, if you can confirm
> earlier alpha and beta builds work for you, then I may be able to see what
> the difference is.
> 
> 
> Curtis Elliott
> […](#)

It does cause an infinite loading in version v2.0.0rc, while version v2.0.0b2 works normally.

**Comment by itsjustcurtis** [comment]
> > They also mentioned that it stopped working in 2.0.0rc, if you can confirm
> > earlier alpha and beta builds work for you, then I may be able to see what
> > the difference is.
> > 
> > 
> > Curtis Elliott
> > […](#)
> 
> It does cause an infinite loading in version v2.0.0rc, while version v2.0.0b2 works normally.

That makes a lot of sense, a bunch of memory changes happened then to fix gtao vehicle despawning. Its likely that those patterns aren't compatible with fivem. I don't think there is much I can do about that without getting rid of that functionality

**Comment by lucienlmy** [comment]
> > > They also mentioned that it stopped working in 2.0.0rc, if you can confirm
> > > earlier alpha and beta builds work for you, then I may be able to see what
> > > the difference is.
> > > Curtis Elliott
> > > […](#)
> > 
> > 
> > It does cause an infinite loading in version v2.0.0rc, while version v2.0.0b2 works normally.
> 
> That makes a lot of sense, a bunch of memory changes happened then to fix gtao vehicle despawning. Its likely that those patterns aren't compatible with fivem. I don't think there is much I can do about that without getting rid of that functionality

A switch can be added in the configuration. If it is turned off, this function will not be used, and at least it can run in this way.
